### ISSUE TITLE
Failure with numerous tasks tied to a single machine

### ISSUE THREAD

**Issue by kholbrook1303** [cuckoobox/cuckoo#599]
I am finding that when submitting numerous concurrent submissions, I receive a DEBUG error:

2015-07-07 11:27:57,785 [lib.cuckoo.core.database] DEBUG: Database error getting available machines: QueuePool limit of size 5 overflow 10 reached, connection timed out, timeout 30

Shortly therafter, I eventually get this prior to machine completion:

Traceback (most recent call last):
  File "./cuckoo.py", line 104, in <module>
    cuckoo_main(max_analysis_count=args.max_analysis_count)
  File "./cuckoo.py", line 78, in cuckoo_main
    sched.start()
  File "/opt/cuckoo/lib/cuckoo/core/scheduler.py", line 538, in start
    for machine in self.db.get_available_machines():
TypeError: 'int' object is not iterable

**Comment by SpoonBoy** [comment]
Are you using default database ? (SQLite)

**Comment by kholbrook1303** [comment]
No.  I had switched due to a message stating that it was recommended to use postgres, or mysql, when running analysis in parallel, due to possible issues.  That said... I reverted the configuration from MySQL to postgres just to test, and same issue... so then i reverted to SQLlite, and the problem seems to have gone away.

So that lead me to looking more into the error, and from what gather, increasing the pool size and max overflow when using the create_engine function with SQLAlchemy by passing the arguments pool_size=INT & max_overflow=INT, may potentially resolve the issue?

**Comment by KillerInstinct** [comment]
Curious, how many machines are you running/loading? How many concurrent tasks at once before you hit this debug message?

**Comment by kholbrook1303** [comment]
In my scenario, I am running 4 total physical machines (Using a modified FOG/Deep Freeze/Power appliance config), but only submitting to one, 5 total submissions (4 Queued for single machine availability).  

In case your wandering, the reason for sending them up this way is that I have a wide-range of configs for different scenarios so I often will resort to a primary config, and go to others if there is either a rootkit type behavior of something that I do not want to restore on reboot for.... WinXP (No Deep Freeze); Win7 (32bit); Win7 2 (32bit); Win7 (64bit) with no injection.

**Comment by jbremer** [comment]
@kholbrook1303 Do you have any updates on this issue? Perhaps you've fixed it by now? Based on your traceback something is definitely going wrong - you didn't modify any of the code, did you?

In the following paste (a snippet from your traceback) `self.db` should *never* be an integer, so.. :)
```python
File "/opt/cuckoo/lib/cuckoo/core/scheduler.py", line 538, in start
for machine in self.db.get_available_machines():
TypeError: 'int' object is not iterable
```

**Comment by jbremer** [comment]
@kholbrook1303 Edit. I'm not sure about your `QueuePool` things, but the exception that you reported has already been fixed upstream in the latest `master` branch, as of 844653565873a9543e0a64838f64610f53a9775e.
Closing this issue as it has already been resolved - please open a new issue if you find any other issues! :)
